src/app/shell/personal-cabinet/provider/create-provider/create-contacts-form/create-contacts-form.component.ts

```diff
@@ -23,36 +23,61 @@ export class CreateContactsFormComponent implements OnInit {
       buildingNumber: new FormControl('', [Validators.required, Validators.pattern(TEXT_WITH_DIGITS_REGEX)]),
       city: new FormControl('', [Validators.required, Validators.pattern(TEXT_REGEX)]),
       district: new FormControl('', [Validators.required, Validators.pattern(TEXT_REGEX)]),
-      region: new FormControl('', [Validators.required, Validators.pattern(TEXT_REGEX)])
+      region: new FormControl('', [Validators.required, Validators.pattern(TEXT_REGEX)]),
     });
 
     this.ActualAddressFormGroup = this.formBuilder.group({
       street: new FormControl('', [Validators.required, Validators.pattern(TEXT_REGEX)]),
       buildingNumber: new FormControl('', [Validators.required, Validators.pattern(TEXT_WITH_DIGITS_REGEX)]),
       city: new FormControl('', [Validators.required, Validators.pattern(TEXT_REGEX)]),
       district: new FormControl('', [Validators.required, Validators.pattern(TEXT_REGEX)]),
-      region: new FormControl('', [Validators.required, Validators.pattern(TEXT_REGEX)])
+      region: new FormControl('', [Validators.required, Validators.pattern(TEXT_REGEX)]),
     });
   }
 
   ngOnInit(): void {
     this.passActualAddressFormGroup.emit(this.ActualAddressFormGroup);
     this.passLegalAddressFormGroup.emit(this.LegalAddressFormGroup);
-    this.isSameAddressControl.valueChanges.subscribe((isSame: boolean) => {
-      (isSame) ? this.ActualAddressFormGroup.patchValue(this.LegalAddressFormGroup.value) : this.ActualAddressFormGroup.reset();
-    });
+    this.initDisableIsSameAddressControl();
 
     this.provider && this.activateEditMode();
   }
 
   activateEditMode(): void {
+    this.LegalAddressFormGroup.addControl('id', this.formBuilder.control(''));
+    this.ActualAddressFormGroup.addControl('id', this.formBuilder.control(''));
+
+    this.isSameAddressControl.setValue(!Boolean(this.provider.actualAddress));
+
     this.LegalAddressFormGroup.patchValue(this.provider.legalAddress, { emitEvent: false });
+    this.provider.actualAddress && this.ActualAddressFormGroup.patchValue(this.provider.actualAddress, { emitEvent: false });
+  }
 
-    if (this.provider?.actualAddress) {
-      this.ActualAddressFormGroup.patchValue(this.provider.actualAddress, { emitEvent: false });
-    } else {
-      this.ActualAddressFormGroup.patchValue(this.provider.legalAddress, { emitEvent: false });
-    }
+  /**
+   * This method makes input enable if radiobutton value is true and sets the value to the formgroup
+   */
+  initDisableIsSameAddressControl(): void {
```

Comment:
I would name it as sameAddressHandler
very confusing name

Reply:
done